monkey/monkey_island/cc/repository/agent_event_encryption.py

```diff
@@ -0,0 +1,69 @@
+import json
+from typing import Callable, Iterable
+
+from common.agent_event_serializers import JSONSerializable
+from common.agent_events import AbstractAgentEvent
+
+ENCRYPTED_PREFIX = "encrypted_"
+
+
+def get_fields_to_encrypt(event: AbstractAgentEvent):
+    """
+    Get the fields of the event that are not part of the base AbstractAgentEvent.
+    """
+    return set(vars(AbstractAgentEvent)["__fields__"].keys()) ^ set(event.dict().keys())
```

Comment:
Is there a reason not to hide this behind `encrypt_event()`. In other words, could we have `encrypt_event()` call this instead of putting the burden on whoever is calling `encrypt_event()`?

Reply:
As discussed, I put in a workaround for this